### PR TITLE
EDU-498 Allow private storage deletion for owners

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-498

### CODE REVIEW

File: src/components/pages/lesson.js [hunk 1]
```diff
@@ -238,14 +247,19 @@ function Lesson({ PageTemplate, initialState }) {
...
+  const privateStorage = isPrivateRoom && !!room.owner.storage?.plan
     ? {
+      usedBytes: room.owner.storage.usedBytes,
+      maxBytes: room.owner.storagePlan.maxBytes,
       rootPath: `rooms/${room._id}/media`,
       initialPath: `rooms/${room._id}/media`,
-      uploadPath: `rooms/${room._id}/media`
+      uploadPath: `rooms/${room._id}/media`,
+      isDeletionEnabled: isRoomOwner || isRoomCollaborator,
+      onUsedBytesUpdated: handleUsedBytesUpdated
```
Comment (adrianaluchian) [inline src/components/pages/lesson.js:262]: I opted for putting this callback here to be less intrusive with this patch fix 😬

File: src/domain/private-data.js [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-
-export function getAllowedUserFields(forUser) {
-  return hasUserPermission(forUser, permissions.SEE_USER_EMAIL)
-    ? extendedFields
-    : basicFields;
-}
-
-export default {
-  getAllowedUserFields
-};
```
Comment (ahelmberger) [inline src/domain/private-data.js:14]: :+1:

File: src/components/pages/lesson.js (hunk 1 above)
Comment (ahelmberger) [reply src/components/pages/lesson.js:287]: Good idea to not pollute all components in between!

File: src/server/lesson-controller.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ const { NotFound, BadRequest, Forbidden, Unauthorized } = httpErrors;
 
 const isRoomOwnerOrCollaborator = ({ room, userId }) => {
   const isOwner = room.owner === userId;
-  const isCollaborator = room.lessonsMode === ROOM_LESSONS_MODE.collaborative && room.members.find(m => m.userId === userId);
+  const isCollaborator = room.lessonsMode === ROOM_LESSONS_MODE.collaborative && room.members.some(m => m.userId === userId);
```
Comment (ahelmberger) [inline src/server/lesson-controller.js:29]: The last place to use `find` would be line 67 in this file ;-)

**Review by ahelmberger — APPROVED** [review]

File: src/components/pages/lesson.js (hunk 1 above)
Comment (adrianaluchian) [reply src/components/pages/lesson.js:287]: Was not original :D

File: src/server/lesson-controller.js (hunk 3 above)
Comment (adrianaluchian) [reply src/server/lesson-controller.js:29]: As in 'it should be `some` there too'? If so, updated, if not, I need more info 🙏

Comment (ahelmberger) [reply src/server/lesson-controller.js:29]: Yes exactly (although not really very important), thanks!

Comment (adrianaluchian) [reply src/server/lesson-controller.js:29]: I think it's important cause we could end up with surprises if we rely on a boolean value. Thanks for pointing it out.

**Pull request merged**